litr/src/main/java/com/linkedin/android/litr/io/ExternalMediaSource.kt

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2022 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ *
+ * Author: Ian Bird
+ */
+package com.linkedin.android.litr.io
+
+import android.media.MediaCodec
+import android.media.MediaFormat
+import android.util.Log
+import android.view.Surface
+import com.linkedin.android.litr.codec.Decoder
+import com.linkedin.android.litr.codec.Frame
+import com.linkedin.android.litr.codec.PassthroughDecoder
+import com.linkedin.android.litr.exception.MediaSourceException
+import java.nio.ByteBuffer
+
+private const val TAG = "ExternalMediaSource"
+private const val DECODER_NAME = "ExternalMediaSource.Decoder"
+
+private const val MIME_TYPE_RAW = "video/raw"
+private const val DEFAULT_FRAME_RATE = 30
+private const val DEFAULT_BITRATE = 10_000_000
+private const val DEFAULT_KEY_FRAME_INTERVAL = 5
+
+/**
+ * An implementation of MediaSource, which exposes the input {@link Surface} of the encoder via a
+ * Callback. An instance of this class is expected to be both the MediaSource, and Decoder for which
+ * the pipeline is configured. This allows these components to be bypassed.
+ */
+open class ExternalMediaSource: MediaSource, Decoder {
```

Comment:
This is a question, not a change suggestion. Would it make sense to name this `CaptureMediaSource` or something similar?

Reply:
Yeah, I like it. I think "External" made more sense when we expected most consumers would need it directly. However, with the addition of `Camera2MediaSource`, most people may use this instead. Will rename